pkg/util/pullsecret/pullsecret.go

```diff
@@ -164,3 +169,41 @@ func Build(oc *api.OpenShiftCluster, ps string) (string, error) {
 
 	return pullSecret, nil
 }
+
+type registryClient struct {
+	checkAuth func(context.Context, *types.SystemContext, string, string, string) error
+}
+
+func NewRegistryClient() registryClient {
+	return registryClient{
+		checkAuth: dockerregistry.CheckAuth,
+	}
+}
+
+// ValidatePullSecret validates a passed in pull secret by attempting to log in to the registry
+func (r *registryClient) ValidatePullSecret(ctx context.Context, secret *corev1.Secret) error {
```

Comment:
+1 on keeping this in util - we could potentially configure a monitor that executes this as well if we decide we need that later.